src/VDPRegViewer.cpp

```diff
@@ -905,17 +912,26 @@ void VDPRegViewer::connectHighLights()
 	makeGroup(list, label_dec_r26);
 }
 
+static constexpr unsigned TOTAL_SIZE = 64 + 16 + 2 + 3;
+
 void VDPRegViewer::refresh()
 {
-	//new SimpleHexRequest("{VDP regs}",0,64,regs, *this);
-	//new SimpleHexRequest("{VDP status regs}",0,16,regs, *this);
-	// now combined in one request:
-	new SimpleHexRequest(
+	bool registerLatchAvailable = VDPDataStore::instance().getRegisterLatchAvailable();
+	bool paletteLatchAvailable = VDPDataStore::instance().getPaletteLatchAvailable();
+	bool dataLatchAvailable = VDPDataStore::instance().getDataLatchAvailable();
+
+	// three to six different requests now combined in a single one:
+	QString req = QString(
 		"debug_bin2hex "
 		"[ debug read_block {VDP regs} 0 64 ]"
 		"[ debug read_block {VDP status regs} 0 16 ]"
-		"[ debug read_block {VRAM pointer} 0 2 ]",
-		64 + 16 + 2, regs, *this);
+		"[ debug read_block {VRAM pointer} 0 2 ]%1%2%3")
+		.arg(registerLatchAvailable ? "[debug read_block {VDP register latch status} 0 1]" : "")
+		.arg(paletteLatchAvailable ? "[debug read_block {VDP palette latch status} 0 1]" : "")
+		.arg(dataLatchAvailable ? "[debug read_block {VDP data latch value} 0 1]" : "");
+
+	int total = TOTAL_SIZE - !registerLatchAvailable - !paletteLatchAvailable - !dataLatchAvailable;
+	new SimpleHexRequest(req, total, regs, *this);
```

Comment:
This is not a comment on your current patch, but on the existing debugger code:
This code block is largely repeated between `VDPDataStore` and `VDPRegViewer`. Can we (in a separate commit) reduce this duplication? Ideally store/query this information only in a single location.